### ISSUE TITLE
Async refresh ui-select: selected options appears in the list again after type in input

### ISSUE THREAD

**Issue by Astray-git** [angular-ui/ui-select#580]
I have a select with a refresh method, 
If I select one item in the list, it will be removed from the options list.
but after I type in the input, the list get refreshed, selected options appears again in the list.

How to avoid this?

**Comment by MBentley1976** [comment]
I've just hit the same exact issue.  Please tell me that I don't have to pre-filter the results before sending them back to the control.

**Comment by MBentley1976** [comment]
Quick follow-up to this.  For now I'm just filtering my results when I get the data back from the ajax call.  


I pass in the select control to the refresh function and grab the selected values from there.
```html
<ui-select-choices repeat="solicitor in locals.solicitorSearchResults | filter: $select.search" refresh="locals.solicitorSearch($select.search, $select)" refresh-delay="100">
```

Then I filter the results based on what's currently selected.
```javascript
.....
.....
var results = result.data.results,
    filteredItems,
    selected = locals.solicitorSearchControl.selected;

filteredItems = results.filter(function (i) {
    return selected.map(function (e) { return e.id; }).indexOf(i.id) < 0;
});

$scope.locals.solicitorSearchResults = filteredItems;
.....
.....
```

**Comment by nmarshall23** [comment]
@MBentley1976 could you go into more detail. That I have to write a filter for this, seems odd.

This feels like a bug in ui-select.

**Comment by MBentley1976** [comment]
Hey Marshall, the issue is the IndexOf() function on the array to look for existing items.  That doesn't work well when trying to compare objects which is what I'm populating my items array with.

**Comment by HS368** [comment]
very simple! just keep reference of selected items and exchange it by duplicate received items
...
if (ctrl.multiple) {
	var selinx = ctrl.selected.map(function (e) { return e.id; }), i, t;
	angular.forEach(items, function(v, k) { 
		if ((i = selinx.indexOf(v.id)) >= 0) {
			t = items[k] = select.selected[i];
			angular.forEach(v, function(n, p) { t[p] = v[p]; });
		}
	});
}
...
---------
<ui-select-choices repeat="itm in items ...

**Comment by wesleycho** [comment]
Going to close as a usage issue - this is a problem that should be solved on the user's side.

**Comment by MattiJarvinen-BA** [comment]
@wesleycho seems to be odd that it should be solved on the user's side instead of UI-select, especially this isn't documented.

**Comment by user378230** [comment]
@MattiJarvinen-BA This has actually been resolved in #1473 from what I can see (Ref: [src/uiSelectController.js#L242](https://github.com/angular-ui/ui-select/blob/fe0c0c12522d48d677fff8ca21d366ba37d8f90d/src/uiSelectController.js#L242))

**Comment by MattiJarvinen-BA** [comment]
@user378230 oh, seems so. So nevermind then. ;)
